applications/StructuralMechanicsApplication/custom_elements/base_shell_element.cpp

```diff
@@ -0,0 +1,629 @@
+{
+    // TODO unify implementation and move it to BaseClass
+}
+
+void BaseShellElement::CalculateDampingMatrix(
+    MatrixType& rDampingMatrix,
+    ProcessInfo& rCurrentProcessInfo
+    )
+{
+    KRATOS_TRY;
+
+    if ( rDampingMatrix.size1() != mNumDofs )
+        rDampingMatrix.resize( mNumDofs, mNumDofs, false );
+
+    noalias( rDampingMatrix ) = ZeroMatrix( mNumDofs, mNumDofs );
+
+    // 1.-Calculate StiffnessMatrix:
+
+    MatrixType StiffnessMatrix  = Matrix();
+    VectorType ResidualVector  = Vector();
+
+    CalculateAll(StiffnessMatrix, ResidualVector, rCurrentProcessInfo, true, false);
+
+    // 2.-Calculate MassMatrix:
+
+    MatrixType MassMatrix  = Matrix();
+
+    CalculateMassMatrix(MassMatrix, rCurrentProcessInfo);
+
+    // 3.-Get Damping Coeffitients (RAYLEIGH_ALPHA, RAYLEIGH_BETA)
+    double alpha = 0.0;
+    if( GetProperties().Has(RAYLEIGH_ALPHA) )
+        alpha = GetProperties()[RAYLEIGH_ALPHA];
+    else if( rCurrentProcessInfo.Has(RAYLEIGH_ALPHA) )
+        alpha = rCurrentProcessInfo[RAYLEIGH_ALPHA];
+
+    double beta  = 0.0;
+    if( GetProperties().Has(RAYLEIGH_BETA) )
+        beta = GetProperties()[RAYLEIGH_BETA];
+    else if( rCurrentProcessInfo.Has(RAYLEIGH_BETA) )
+        beta = rCurrentProcessInfo[RAYLEIGH_BETA];
+
+    // 4.-Compose the Damping Matrix:
+
+    // Rayleigh Damping Matrix: alpha*M + beta*K
+    noalias( rDampingMatrix ) += alpha * MassMatrix;
+    noalias( rDampingMatrix ) += beta  * StiffnessMatrix;
+
+    KRATOS_CATCH( "" )
+}
+
+int BaseShellElement::Check(const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY;
+
+    Element::Check(rCurrentProcessInfo);
+
+    CheckVariables();
+    CheckDofs();
+    CheckProperties(rCurrentProcessInfo);
+
+    KRATOS_ERROR_IF(GetGeometry().Area() < std::numeric_limits<double>::epsilon()*1000)
+        << "Element #" << Id() << " has an Area of zero!" << std::endl;
+
+    // TODO check ConstLaws
+
+    return 0;
+
+    KRATOS_CATCH( "" )
+}
+
+///@}
+///@name Access
+///@{
+
+
+///@}
+///@name Inquiry
+///@{
+
+
+///@}
+///@name Input and output
+///@{
+
+/// Turn back information as a string.
+
+std::string BaseShellElement::Info() const {
+  std::stringstream buffer;
+  buffer << "BaseShellElement #" << Id();
+  return buffer.str();
+}
+
+/// Print information about this object.
+
+void BaseShellElement::PrintInfo(std::ostream& rOStream) const {
+  rOStream << "BaseShellElement #" << Id();
+}
+
+/// Print object's data.
+
+void BaseShellElement::PrintData(std::ostream& rOStream) const {
+  pGetGeometry()->PrintData(rOStream);
+}
+
+///@}
+///@name Friends
+///@{
+
+///@}
+
+///@name Protected static Member Variables
+///@{
+
+///@}
+///@name Protected member Variables
+///@{
+
+///@}
+///@name Protected Operators
+///@{
+
+///@}
+///@name Protected Operations
+///@{
+
+void BaseShellElement::SetBaseMembers()
+{
+    mNumDofs = 6 * GetGeometry().PointsNumber(); // 6 dofs per node
```

Comment:
Do you really need a member variable of something that you always know and can compute in any moment?